djangocms_versioning/admin.py

```diff
@@ -16,23 +24,45 @@ def save_model(self, request, obj, form, change):
         super().save_model(request, obj, form, change)
         if not change:
             # create a new version object and save it
-            extension = apps.get_app_config('djangocms_versioning').cms_extension
-            version_model = extension.content_to_version_models[obj.__class__]
-            version_model.objects.create(content=obj)
+            Version.objects.create(content=obj)
 
     def get_queryset(self, request):
         """Limit query to most recent content versions
         """
         queryset = super().get_queryset(request)
-        versioning_extension = apps.get_app_config(
-            'djangocms_versioning').cms_extension
-        version_model = versioning_extension.content_to_version_models[
-            queryset.model]
-        filter_name = '{}__in'.format(version_model.__name__.lower())
-        latest_versions = version_model.objects.distinct_groupers()
-        return queryset.filter(**{filter_name: latest_versions})
+        versioning_extension = apps.get_app_config('djangocms_versioning').cms_extension
+        versionable = versioning_extension.versionables_by_content[queryset.model]
+        return queryset.filter(pk__in=versionable.distinct_groupers())
 
 
+class VersionChangeList(ChangeList):
```

Comment:
Maybe not a discussion for this PR, but I think this is becoming quite complex and am wondering if it wouldn't be simpler for maintainability to create a custom view for the version list rather than modify the django changelist for this purpose. In my admin experiences so far, it's been the case that as an application grows, a custom view turns out to be much easier to maintain.

Reply:
@czpython ?

Reply:
@monikasulik Talked to Paulo, custom view would require maintaining admin UI, so that breadcrumbs would still have to work, filtering and sorting would have to be reimplemented. So for now, we're sticking with regular changelist view